roles/rhsm-subscription/tasks/main.yaml

```diff
@@ -30,6 +35,6 @@
         password: "{{ rhsm_password }}"
         state: present
         pool: "{{ rhsm_pool }}"
-      when: rhsm_user is defined and rhsm_user
+      when: "'Current' not in subscribed.stdout and rhsm_user is defined and rhsm_user"
```

Comment:
@pschiffe and @dav1x I think this should work for skipping the step rather than Subscribed where we ran into "Not Subscribed"